### ISSUE TITLE
[FEATURE]  OUTPUT needed so i can record the guids of new records to create relationships

### ISSUE THREAD

**Issue by magiva** [MarkMpn/Sql4Cds#597]
**Is your feature request related to a problem? Please describe.**
during an update or insert i want to get the record id of each new row so i can then update the source data rows, essentially creating a relationship to the new records.

**Describe the solution you'd like**
INSERT INTO target_table (name, address)
OUTPUT inserted.row_guid, source_table.source_guid

**Describe alternatives you've considered**
i used creation date to see the records id created and then used that data to manually align (it was late :))

**Additional context**

**Comment by MarkMpn** [comment]
Thanks for the suggestion!

You can get the ID of newly inserted records using `@@IDENTITY` which might get you what you need, e.g.

```sql
INSERT INTO account (name) VALUES ('New Account')

INSERT INTO contact (firstname, lastname, parentcustomerid) VALUES ('Mark', 'Carrington', @@IDENTITY)
```

**Comment by magiva** [comment]
its good idea but i thought this is only for 1 record as it returns only the last insert.
wouldnt work for multiple inserts i believe.  happy to be wrong though :)
i think my better option would be to create a temp table in dataverse to handle it

**Comment by MarkMpn** [comment]
Indeed, using `OUTPUT` would be a good enhancement but flagging `@@IDENTITY` as a possible workaround for now when you are working with single records.

**Comment by magiva** [comment]
appreciate the response

**Comment by marcotidona** [comment]
Something like this would be wonderful so we could generate and re-play mock generation:

DECLARE @newAccountId UNIQUEIDENTIFIER;

INSERT INTO account (name) 
OUTPUT INSERTED.accountid INTO @newAccountId
VALUES ('Test Account');

-- Now use this ID in a related record
INSERT INTO contact (firstname, lastname, parentcustomerid) 
VALUES ('John', 'Doe', @newAccountId);

Thank you for the great work!

Cheers,
Marco
